ADAL/src/request/ADAuthenticationRequest+Broker.m

```diff
@@ -306,14 +307,25 @@ - (void)callBroker:(ADAuthenticationCallback)completionBlock
     id<ADTokenCacheDataSource> dataSource = [_tokenCache dataSource];
     if (dataSource && [dataSource isKindOfClass:[ADKeychainTokenCache class]])
     {
+        NSString* keychainGroup = [(ADKeychainTokenCache*)dataSource sharedGroup];
+        NSString* teamId = [ADKeychainUtil keychainTeamId:&error];
```

Comment:
+[ADKeychainUtil retrieveTeamIDFromKeychain] will create the entry it is looking for if it doesn't exist.

And, no, keychainTeamId will not return an empty string and an error either. The error it returns is from retrieveTeamIDFromKeychain when they keychain operation fails. In that case the keychainTeamID never gets set.